File: backend/src/main/java/com/gittowork/domain/user/entity/UserGitInfo.java
package com.gittowork.domain.user.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.Instant;
import java.time.LocalDateTime;


@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserGitInfo {

    @Id
    @Column(name = "user_id", nullable = false)
    private Integer id;

    @MapsId
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Size(max = 255)
    @NotNull
    @Column(name = "avatar_url", nullable = false)
    private String avatarUrl;

    @NotNull
    @Column(name = "public_repositories", nullable = false)
    private Integer publicRepositories;

    @NotNull
    @Column(name = "followers", nullable = false)
    private Integer followers;

    @NotNull
    @Column(name = "followings", nullable = false)
    private Integer followings;

    @NotNull
    @Column(name = "create_dttm", nullable = false)
    private Instant createDttm;

    @Column(name = "update_dttm")
    private Instant updateDttm;

}

File: bigdata/app/models/user_git_info.py
from sqlalchemy import Column, Integer, String, TIMESTAMP, ForeignKey
from sqlalchemy.orm import relationship
from app.core.database import Base

class UserGitInfo(Base):
    __tablename__ = "user_git_info"

    user_id = Column(Integer, ForeignKey("user.user_id"), primary_key=True)
    avatar_url = Column(String(255), nullable=False)
    public_repositories = Column(Integer, nullable=False)
    followers = Column(Integer, nullable=False)
    followings = Column(Integer, nullable=False)
    create_dttm = Column(TIMESTAMP, nullable=False)
    update_dttm = Column(TIMESTAMP, nullable=True)

    # 관계
    user = relationship("User", back_populates="user_git_info")
